src/trading/Orders.sol

```diff
@@ -0,0 +1,320 @@
+/**
+ * Copyright (C) 2015 Forecast Foundation OU, full GPL notice in LICENSE
+ */
+
+pragma solidity ^0.4.13;
+
+import 'ROOT/Controller.sol';
+import 'ROOT/libraries/math/SafeMathInt256.sol';
+import 'ROOT/libraries/Trading.sol';
+import 'ROOT/reporting/Interfaces.sol';
+import 'ROOT/trading/OrdersFetcher.sol';
+
+
+/**
+ * @title Orders
+ * @dev Storage of all data associated with orders
+ */
+contract Orders is Controlled {
+    using SafeMathUint256 for uint256;
+
+    event CancelOrder(address indexed market, address indexed sender, int256 fxpPrice, uint256 fxpAmount, bytes20 orderId, uint8 outcome, Trading.TradeTypes orderType, uint256 cashRefund, uint256 sharesRefund);
+    event CompleteSets(address indexed sender, address indexed market, Trading.TradeTypes indexed orderType, uint256 fxpAmount, uint256 numOutcomes, uint256 marketCreatorFee, uint256 reportingFee);
+    event MakeOrder(address indexed market, address indexed sender, Trading.TradeTypes indexed orderType, int256 fxpPrice, uint256 fxpAmount, uint8 outcome, bytes20 orderId, uint256 fxpMoneyEscrowed, uint256 fxpSharesEscrowed, uint256 tradeGroupId);
+    event TakeOrder(address indexed market, uint8 indexed outcome, Trading.TradeTypes indexed orderType, bytes20 orderId, int256 price, address maker, address taker, uint256 makerShares, int256 makerTokens, uint256 takerShares, int256 takerTokens, uint256 tradeGroupId);
+
+    struct Order {
+        uint256 fxpAmount;
+        int256 fxpPrice;
+        address owner;
+        uint256 fxpSharesEscrowed;
+        uint256 fxpMoneyEscrowed;
+        bytes20 betterOrderId;
+        bytes20 worseOrderId;
+    }
+
+    struct MarketOrders {
+        uint256 volume;
+        mapping(uint8 => int256) prices;
+    }
+
+    mapping(bytes20 => Order) private orders;
+    mapping(address => MarketOrders) private marketOrderData;
+    mapping(bytes20 => bytes20) private bestOrder;
+    mapping(bytes20 => bytes20) private worstOrder;
+
+    // Getters
+    function getOrders(bytes20 _orderId, Trading.TradeTypes _type, Market _market, uint8 _outcome) public constant returns (uint256, int256, address, uint256, uint256, bytes20, bytes20) {
+        Order _order = orders[_orderId];
+        return (_order.fxpAmount, _order.fxpPrice, _order.owner, _order.fxpSharesEscrowed, _order.fxpMoneyEscrowed, _order.betterOrderId, _order.worseOrderId);
+    }
+
+    function getBestOrders(Trading.TradeTypes _type, Market _market, uint8 _outcome) public constant returns (bytes20) {
+        return bestOrder[getBestOrderWorstOrderHash(_market, _outcome, _type)];
+    }
+
+    function getWorstOrders(Trading.TradeTypes _type, Market _market, uint8 _outcome) public constant returns (bytes20) {
+        return worstOrder[getBestOrderWorstOrderHash(_market, _outcome, _type)];
+    }
+
+    function getAmount(bytes20 _orderId, Trading.TradeTypes _type, Market _market, uint8 _outcome) public constant returns (uint256) {
```

Comment:
Solc is complaining about unused variables for a lot of these.  I know we are keeping them in for now so we don't have to change too much, but we should remove the parameter names and just have `(bytes20 _orderId, Trading.TradeTypes, Market, uint8)` as the parameter list so solc understands that we intentionally aren't using them.